src/dns_server.c

```diff
@@ -53,40 +53,74 @@ Contains the freeRTOS task for the DNS server that processes the requests.
 #include "attribs.h"
 #include "os_task.h"
 #include "log.h"
+#include "esp_type_wrapper.h"
 
-static const char   TAG[]           = "dns_server";
-static TaskHandle_t task_dns_server = NULL;
-int                 socket_fd;
+static const char TAG[] = "dns_server";
+
+static TaskHandle_t gh_dns_task;
+static socket_t     g_dns_socket_fd = -1;
 
 ATTR_NORETURN
 void
-dns_server(ATTR_UNUSED void *p_params);
+dns_server_task(ATTR_UNUSED void *p_params);
 
-void
+bool
 dns_server_start(void)
 {
+    /* Create UDP socket */
+    socket_t socket_fd = socket(AF_INET, SOCK_DGRAM, 0);
+    if (socket_fd < 0)
```

Comment:
can `socket()` return other values than `-1` if socket creation fails? If no, can we check `if (SOCKET_NOT_INIT == socket)`?  If yes, please amend above comment. For example ` Create UDP socket - returns negative value on error`.